### PR TITLE
Convert optimistic challenges hooks into selector

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Converts the audio rewards page hooks into a selector. Avoids problems with importing them on mobile, and also seems to conceptually make sense.

Also moves the types to the proper models file.

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

- Removes the `useOptimisticChallenge` hook, as it seems it wasn't being used.

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-hooks-to-selectors

File: src/pages/audio-rewards-page/ChallengeRewardsTile.tsx [hunk 1]
```diff
@@ -59,11 +62,11 @@ const RewardPanel = ({
   stepCount
 }: RewardPanelProps) => {
   const wm = useWithMobileStyle(styles.mobile)
-  const userChallenges = useSelector(getUserChallenges)
+  const userChallenges = useSelector(getOptimisticUserChallenges)
```
Comment (piazzatron) [inline src/pages/audio-rewards-page/ChallengeRewardsTile.tsx:65]: Could also have a second version of this selector that takes in an id, and calls this one and indexes into it

**Review by piazzatron — APPROVED** [review]
Looks gr8

File: src/pages/audio-rewards-page/ChallengeRewardsTile.tsx (hunk 1 above)
Comment (rickyrombo) [reply src/pages/audio-rewards-page/ChallengeRewardsTile.tsx:65]: yeah, we had something similar before in the hooks, but it wasn't getting used so I figured i'd not bring it over

**Pull request merged**